pkg/cryptutil/jose.go

```diff
@@ -93,3 +97,15 @@ func loadPublicKey(b []byte) (interface{}, error) {
 
 	return nil, fmt.Errorf("couldn't load public key: %w", wrappedErr)
 }
+
+// SignatureAlgorithmForKey returns the signature algorithm for the given key.
+func SignatureAlgorithmForKey(key interface{}) (jose.SignatureAlgorithm, error) {
```

Comment:
Could we add a few unit tests that explicitly test this function for both private and public keys for both algo types?